biodatasets/swedish_medical_ner/swedish_medical_ner.py

```diff
@@ -0,0 +1,290 @@
+
+It consists three subsets which are in turn derived from three different sources respectively: 
+
+* the Swedish Wikipedia (a.k.a. wiki): Wiki_annotated_60.txt
+* Läkartidningen (a.k.a. lt): LT_annotated_60.txt
+* 1177 Vårdguiden (a.k.a. 1177): 1177_annotated_sentences.txt
+
+Texts from both Swedish Wikipedia and Läkartidningen were automatically annotated using a 
+list of medical seed terms. Sentences from 1177 Vårdguiden were manuually annotated.
+
+It can be found in Hugging Face Datasets: https://huggingface.co/datasets/swedish_medical_ner.
+
+"""
+
+import os
+from typing import Dict, List, Tuple
+
+import datasets
+import regex as re
+
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_DATASETNAME = "swedish_medical_ner"
+
+_CITATION = """\
+@inproceedings{almgren-etal-2016-named,
+    author = {
+        Almgren, Simon and
+        Pavlov, Sean and
+        Mogren, Olof
+    },
+    title     = {Named Entity Recognition in Swedish Medical Journals with Deep Bidirectional Character-Based LSTMs},
+    booktitle = {Proceedings of the Fifth Workshop on Building and Evaluating Resources for Biomedical Text Mining (BioTxtM 2016)},
+    publisher = {The COLING 2016 Organizing Committee},
+    pages     = {30-39},
+    year      = {2016},
+    month     = {12},
+    url       = {https://aclanthology.org/W16-5104},
+    eprint    = {https://aclanthology.org/W16-5104.pdf}
+}
+"""
+
+_DESCRIPTION = """\
+swedish_medical_ner is Named Entity Recognition dataset on medical text in Swedish. 
+It consists three subsets which are in turn derived from three different sources 
+respectively: the Swedish Wikipedia (a.k.a. wiki), Läkartidningen (a.k.a. lt), 
+and 1177 Vårdguiden (a.k.a. 1177). While the Swedish Wikipedia and Läkartidningen 
+subsets in total contains over 790000 sequences with 60 characters each, 
+the 1177 Vårdguiden subset is manually annotated and contains 927 sentences, 
+2740 annotations, out of which 1574 are disorder and findings, 546 are 
+pharmaceutical drug, and 620 are body structure.
+
+Texts from both Swedish Wikipedia and Läkartidningen were automatically annotated 
+using a list of medical seed terms. Sentences from 1177 Vårdguiden were manuually 
+annotated.
+"""
+
+_HOMEPAGE = "https://github.com/olofmogren/biomedical-ner-data-swedish/"
+
+_LICENSE = "Creative Commons Attribution-ShareAlike 4.0 International Public License (CC BY-SA 4.0)"
+
+_URLS = {
+    _DATASETNAME: [
+        "https://raw.githubusercontent.com/olofmogren/biomedical-ner-data-swedish/master/Wiki_annotated_60.txt",
+        "https://raw.githubusercontent.com/olofmogren/biomedical-ner-data-swedish/master/LT_annotated_60.txt",
+        "https://raw.githubusercontent.com/olofmogren/biomedical-ner-data-swedish/master/1177_annotated_sentences.txt",
+    ],
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+def _read_data(datapaths):
```

Comment:
Comments
- Please document what this function is doing. I believe it's using regular expressions to find the markup indicating entities so that we can assign span offsets? 
- This call runs very slowly for me (>2 minutes) when it should take seconds.  I suspect it is due to all the regex calls. Can you optimize this to run faster? I don't follow what the regex patterns are doing out-of-box, but I suspect you can optimize here easily.

Reply:
Yes to your first question. The original dataset doesn't include entity offsets but brackets around the entities which are used to locate them. 

I will see if I can make the code run faster, the regex calls are necessary for extracting and locating entities while getting rid of the brackets in the sentences. I will add more comments. Btw the dataset has over 700000 sentences.